pkg/streamingpromql/operators/selectors/extend_range_vector.go

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/prometheus/prometheus/blob/main/promql/engine.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Prometheus Authors
+
+package selectors
+
+import (
+	"github.com/prometheus/prometheus/promql"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+	"github.com/grafana/mimir/pkg/util/limiter"
+)
+
+type ExtendedPoints struct {
+	points []promql.FPoint
+}
+
+type ExtendedSmoothedPoints struct {
```

Comment:
Do we need "extended" in the name if this represents smoothed points?

```suggestion
type SmoothedPoints struct {
```

(same applies for `ExtendedSmoothedRangeVectorPoints` below)